examples/vnet/doca_gw.c

```diff
@@ -66,24 +106,24 @@ struct doca_gw_pipelne_entry *doca_gw_pipeline_add_entry(uint16_t pipe_queue,
                       struct doca_gw_actions *actions,struct doca_gw_monitor *mon,
                       struct doca_fwd_tbl *fwd, struct doca_gw_error *err)
 {
-    static int  pipe_entry_id = 0;
+
     struct doca_gw_pipelne_entry *entry;
 
     if(fwd == NULL) {
         DOCA_LOG_WARN("no forwading");
         return NULL;
     }
-    entry = (struct doca_gw_pipelne_entry *) malloc(sizeof(struct doca_gw_pipelne_entry));
+	entry = doca_gw_alloc_new_entry();
     if (entry == NULL)
             return NULL;
-    memset(entry,0,sizeof(struct doca_gw_pipelne_entry));
     entry->pipe_entry = doca_gw_dpdk_pipe_create_flow(entry, pipeline->handler,
 		match, actions, mon, &fwd->cfg, err);
     if (entry->pipe_entry == NULL) {
             DOCA_LOG_INFO("create pip entry fail.\n");
             goto free_pipe_entry;
     }
-    entry->id = pipe_entry_id++;
+    entry->id = pipeline->pipe_entry_id++;
+	LIST_INSERT_HEAD(&pipeline->entry_list, entry, next);
```

Comment:
I think this should go to doca_gw_dpdk layer. you don't need to handle here how you save the entries

Reply:
it's need more code to do it ,  can commit  with another commit ?